src/pages/edit_resource/contents_step/ChangeMainContent.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+import { Button } from '@mui/material';
+import styled from 'styled-components';
+import { Content } from '../../../types/content.types';
+import { DashboardModal, useUppy } from '@uppy/react';
+import { createMainFileChangerUppy, uppyLocale } from '../../../utils/uppy-config';
+import { Resource } from '../../../types/resource.types';
+import {
+  setContentAsDefaultThumbnail,
+  setContentAsDefaultContent,
+  setContentAsMasterContent,
+} from '../../../api/fileApi';
+import { useFormikContext } from 'formik';
+import { resetFormButKeepTouched } from '../../../utils/formik-helpers';
+import { useTranslation } from 'react-i18next';
+import ErrorBanner from '../../../components/ErrorBanner';
+import { handlePotentialAxiosError } from '../../../utils/AxiosErrorHandling';
+
+const ChangeMainFileWrapper = styled.div`
+  margin-top: 1rem;
+`;
+
+const ChangeMainContent = () => {
+  const [showMainFileUppyModal, setShowMainFileUppyModal] = useState(false);
+  const { t } = useTranslation();
+  const { values, resetForm, touched, setTouched } = useFormikContext<Resource>();
+  const [newContent, setNewContent] = useState<Content | undefined>();
+  const mainFileChangeUppy = useUppy(createMainFileChangerUppy(values.identifier, setNewContent));
+  const [uploadError, setUploadError] = useState<Error | undefined>();
```

Comment:
mener typen blir `Xxx | undefined` be default, så man ikke trneger å spesifisere det
```suggestion
  const [newContent, setNewContent] = useState<Content>();
  const mainFileChangeUppy = useUppy(createMainFileChangerUppy(values.identifier, setNewContent));
  const [uploadError, setUploadError] = useState<Error>();
```

Reply:
fikset